src/database/src/output/airspace.rs

```diff
@@ -0,0 +1,185 @@
+                    'G' => PathType::GreatCircle,
+                    'H' | _ => PathType::RhumbLine,
+                },
+            },
+            'L' | 'R' => Self {
+                location: Coordinates {
+                    lat: latitude.unwrap(),
+                    long: longitude.unwrap(),
+                },
+                arc: Some(Arc {
+                    origin: Coordinates {
+                        lat: arc_latitude.unwrap(),
+                        long: arc_longitude.unwrap(),
+                    },
+                    distance: arc_distance.unwrap(),
+                    bearing: arc_bearing.unwrap(),
+                    direction: match boundary_char {
+                        'R' => TurnDirection::Right,
+                        'L' | _ => TurnDirection::Left,
+                    },
+                }),
+                path_type: PathType::Arc,
+            },
+            _ => panic!("Invalid path type"),
+        }
+    }
+}
+
+#[derive(Serialize, Debug)]
+pub struct ControlledAirspace {
+    pub area_code: String,
+    pub icao_code: String,
+    pub airspace_center: String,
+    pub name: String,
+    pub airspace_type: ControlledAirspaceType,
+    pub boundary_paths: Vec<Path>,
+}
+
+#[derive(Serialize, Debug)]
+pub struct RestrictiveAirspace {
+    pub area_code: String,
+    pub icao_code: String,
+    pub designation: String,
+    pub name: String,
+    pub airspace_type: RestrictiveAirspaceType,
+    pub boundary_paths: Vec<Path>,
+}
+
+pub(crate) fn map_controlled_airspaces(data: Vec<sql_structs::ControlledAirspace>) -> Vec<ControlledAirspace> {
+    let mut airspace_complete = false;
+
+    data.into_iter().fold(Vec::new(), |mut airspaces, row| {
+        if airspaces.len() == 0 || airspace_complete {
+            airspaces.push(ControlledAirspace {
+                area_code: row.area_code.clone(),
+                icao_code: row.icao_code.clone(),
+                airspace_center: row.airspace_center.clone(),
+                name: row
+                    .controlled_airspace_name
+                    .clone()
+                    .expect("First row of an airspace data must have a name"),
+                airspace_type: row.airspace_type.clone(),
+                boundary_paths: Vec::new(),
+            });
+
+            airspace_complete = false;
+        }
+
+        if row.boundary_via.chars().nth(1) == Some('E') {
+            airspace_complete = true;
+        }
+
+        let target_airspace = airspaces.last_mut().unwrap();
+
+        target_airspace.boundary_paths.push(Path::from_data(
+            row.latitude,
+            row.longitude,
+            row.arc_origin_latitude,
+            row.arc_origin_longitude,
+            row.arc_distance,
+            row.arc_bearing,
+            row.boundary_via,
+        ));
+
+        airspaces
+    })
+}
+
+pub(crate) fn map_restrictive_airspaces(data: Vec<sql_structs::RestrictiveAirspace>) -> Vec<RestrictiveAirspace> {

```

Comment:
Same applies here

Reply:
Same as before, we dont need to expose internal workings out of the crate